demos/TransferLearning/TransferLearningDemo.py

```diff
@@ -411,86 +220,86 @@ def compile_model(self, modelName, fine=0):
+                                           validation_data=self.validation_generator,
+                                           validation_steps=validation_steps,
+                                           verbose=self.verbose,
+                                           callbacks=callbacks)
+        return history
 
-        if self.verbose:
-            with self.out_stats:
-                wrong = ~(predicted_class_indices ^ self.test_class_indices) + 2
-                print("Correct Matrix")
-                print(wrong)
+    def predict(self, callbacks=None):
+        if self.model:
+            probabilities = self.model.predict_generator(self.test_generator,
+                                                         verbose=self.verbose,
+                                                         workers=self.workers,
+                                                         callbacks=callbacks)
+            predicted_class_indices = np.argmax(probabilities, axis=1)
 
-            print("Total guessed:", wrong.shape[0])
-            print("Accuracy:", np.count_nonzero(wrong) / wrong.shape[0])
+            return probabilities
 
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(prog='TransferLearningDemo')
-    parser.add_argument('--gui', help='shows the GUI of the demo', action='store_true')
     parser.add_argument('--training',
                         help='performs the training phase of the demo',
                         action='store_true')
+    parser.add_argument('--predict',
+                        help='performs the inference phase of the demo',
+                        action='store_true')
     parser.add_argument(
         '--network_type',
         help='selects the network used for training/classification [ResNet50]/MobileNet V2')
     parser.add_argument(
-        '--backend',
-        help=
-        'selects the backend used for training/classification (run ngraph_bridge.list_backends() for full list)'
-    )
+        '--backend', help='selects the backend used for training/classification [CPU]/PLAIDML/TF]')
     parser.add_argument('--quiet', help='disables most logging', action='store_false')
+    parser.add_argument('--epochs', help='number of epochs to train')
+    parser.add_argument('--batch_size', help='specify batch size for training')
+    parser.add_argument('--workers', help='specify number of workers for threading')
+    parser.add_argument('--warmup', help='warmup run for training', action='store_true')
     args = parser.parse_args()
     nw = 'ResNet50'
     if args.network_type == 'ResNet50' or args.network_type == 'MobileNet V2':
         nw = args.network_type
-    be = "PLAIDML"
-    if args.backend != "" and args.backend in ngraph_bridge.list_backends():
+    be = "CPU"
+    if args.backend == 'CPU' or args.backend == 'TF' or args.backend == 'PLAIDML':
         be = args.backend
-    Demo(args.gui, args.training, nw, be, args.quiet)
+    if args.epochs:
```

Comment:
You probably want to check that this value is a positive number

Reply:
While you're at it, you may want to check that `args.batch_size` and `args.workers` are positive numbers as well. I don't think any of this code will play nice if someone passes in a negative value to these args.